### PR TITLE
Added "Why IHP is using Nix"

### PR DESCRIPTION
**Pull request by mpscholten**

### CODE REVIEW

File: blog/2020-07-22-why-ihp-is-using-nix.html [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+            <h2 style="margin-top: 0; margin-bottom: 0rem; font-weight: 900; font-size: 3rem">Why IHP is Using The Nix Package Manager</h1>
+            <p style="margin-top: 1rem; font-size: 1.75rem; font-weight: 600; color:hsla(196, 13%, 80%, 1)">by Marc Scholten, 22.07.2020</p>
+        </div>
+    </div>
+
+    <div class="container">
+        <div style="margin-bottom: 5rem">
+            <div class="intro-text">
+                <p>
+                    IHP is using the Nix Package Manager for managing it dependencies, such as haskell packages, compilers and even postgres. There has been a bit of discussion <a href="https://github.com/digitallyinduced/ihp/issues/131" target="_bl">on GitHub</a>, <a href="https://www.reddit.com/r/haskell/comments/hee4r4/ihp_is_a_modern_batteriesincluded_web_framework/fvr7ste/" target="_blank">reddit</a> and <a href="https://news.ycombinator.com/item?id=23615060" target="_blank">Hacker News</a> around the use of nix. We want to respond to this discussion by sharing a bit of the background on why we decided to make nix a hard requirement for IHP.
```
Comment (hendi) [inline blog/2020-07-22-why-ihp-is-using-nix.html:47]: "for managing **its** dependencies"

File: blog/2020-07-22-why-ihp-is-using-nix.html [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    
+
+    <div class="container">
+        <div>
+            <ul class="features">
+                <li>
+                    <strong>Easy Onboarding</strong>
+
+                    <p>
+                        Making haskell web apps is a complicated process. You need the right haskell compiler, a build tool, lots of libraries to do useful things and a database software. To make IHP as accessible as possible, we want to initially hide all this from the user. We set out to make IHP usable for people who have not yet header about ghc and cabal.
```
Comment (hendi) [inline blog/2020-07-22-why-ihp-is-using-nix.html:62]: "who have not yet **heard** about ghc and cabal."

File: blog/2020-07-22-why-ihp-is-using-nix.html [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+                    </p>
+
+                    <p>
+                        Additionally when you are using an older compiler version, some library is missing or your postgres is not the right version this will cause trouble. Solving these package management issues is always a huge time sink.
+                    </p>
+
+                    <p>
+                        Npm is mainly used for packaging javascript code. Composer is used for packaging php code. And homebrew is used for installing npm and composer ;) Nix is different: It is able to package system software and all kind of software ecosystems. This is allows us to bundle all the dependencies of IHP using just a single package manager. Exactly what we need to make a truly batteries-included framework.
```
Comment (hendi) [inline blog/2020-07-22-why-ihp-is-using-nix.html:70]: "and all **kinds** of software"

Comment (hendi) [inline blog/2020-07-22-why-ihp-is-using-nix.html:70]: "This is **what** allows us to"

File: blog/2020-07-22-why-ihp-is-using-nix.html [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+
+                <li>
+                    <strong>Integrated Dev Tooling</strong>
+
+                    <p>
+                        Nix is unique in that it allows us to exactly specify the version of a package we want to install. We can say: <i>Install the latest <q>postgres 12.3</q></i>. Nix also allows us to install packages locally for a specific project instead of managing all the dependencies system wide.
+                    </p>
+                    
+                    <p>
+                        These two properties make nix a great fit to archive our goal of simplifying the package management process. Nix allows us to easily install all this software for developers working with IHP. We can install exactly the right compiler, the right postgres version as well as all the right libraries we need. We can be sure that all packages are installed in the right version (no time sinks!) and nothing is missing. We also don't have any interference with existing packages already installed on the system. It's like docker without the overhead of a VM.
```
Comment (hendi) [inline blog/2020-07-22-why-ihp-is-using-nix.html:83]: "to **achive** our goal"

File: blog/2020-07-22-why-ihp-is-using-nix.html [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+                    </p>
+
+                    <p>
+                        The IHP dev server automatically starts and manages a postgres server in the background, just for your project. <a href="https://news.ycombinator.com/item?id=23612217" target="_blank">IHP is the first framework tighly integrating the database in the dev mode.</a> <a href="https://twitter.com/zaiste/status/1277678880782979072" target="_blank">We're sure that this will be standard in a couple of years.</a> Without the possibilities of nix, we would have never tried to the integrate postgres into the IHP dev server. The integrated database schema designer wouldn't really work that well. It might not be obvious, but without nix we would not have tools like the IHP schema designer or the database viewer.
```
Comment (hendi) [inline blog/2020-07-22-why-ihp-is-using-nix.html:87]: "tried to <strike>the</strike> integrate postgres"

File: blog/2020-07-22-why-ihp-is-using-nix.html [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+                    </p>
+
+                </li>
+
+                <li>
+                    <strong>Pinned Versions</strong>
+
+                    <p>
+                        Nix makes IHP projects reproducable in a way unseen in other software ecosystems.
+                        <a href="https://github.com/digitallyinduced/ihp-boilerplate/blob/master/Config/nix/nixpkgs-config.nix#L21" target="_blank">All IHP projects share exactly the same set of packages.</a> The same compiler version, the same libraries, the same postgres version and even the same custom dependencies your app depends on. <q>Works on my machine</q> will not happen anymore. When there is a problem, other IHP users will most likely already faced the same issue.
```
Comment (hendi) [inline blog/2020-07-22-why-ihp-is-using-nix.html:97]: "IHP users will most likely already **have** faced the same issue."

**Comment by mpscholten** [comment]
Thanks :) All fixed

**Pull request merged**